packages/astro/src/content/loaders/glob.ts

```diff
@@ -102,6 +103,10 @@ export function glob(globOptions: GlobOptions): Loader {
 				const digest = generateDigest(contents);
 
 				if (existingEntry && existingEntry.digest === digest && existingEntry.filePath) {
+					if (entryType.extensions.includes('.mdx')) {
+						store.addModuleImport(existingEntry.filePath, 'astro:content-module-imports');
```

Comment:
Rather than using the virtual module here, could you instead use the filepath as the specifier? You're not actually checking the virtual module ID inside the Vite plugin, and are actually just using the flag to match it. You can then save `astro:content-module-imports` as just the virtual module to import the map itself.

Reply:
I think the bug is that we don't check for `astro:content-module-imports` **and** the flag, which we should. I can test your suggestion, I am not sure if using the file path as a specifier could work on the long run, because it could trigger other rollup plugin

Reply:
If we do use the virtual module id, then I think it should be a different one from the one used to import the actual manifest.